src/Features/VisualBasic/Portable/ChangeSignature/VisualBasicChangeSignatureService.vb

```diff
@@ -105,25 +107,44 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ChangeSignature
                 Return Nothing
             End If
 
+            Dim insertPosition = TryGetInsertPositionFromDeclaration(matchingNode)
             Dim semanticModel = Await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(False)
             Dim symbol = TryGetDeclaredSymbol(semanticModel, matchingNode, token, cancellationToken)
             If symbol IsNot Nothing Then
                 Dim selectedIndex = TryGetSelectedIndexFromDeclaration(position, matchingNode)
-                Return (symbol, selectedIndex)
+                Return (symbol, selectedIndex, insertPosition)
             End If
 
             If matchingNode.Kind() = SyntaxKind.ObjectCreationExpression Then
                 Dim objectCreation = DirectCast(matchingNode, ObjectCreationExpressionSyntax)
                 If token.Parent.AncestorsAndSelf().Any(Function(a) a Is objectCreation.Type) Then
                     Dim typeSymbol = semanticModel.GetSymbolInfo(objectCreation.Type).Symbol
                     If typeSymbol IsNot Nothing AndAlso typeSymbol.IsKind(SymbolKind.NamedType) AndAlso DirectCast(typeSymbol, ITypeSymbol).TypeKind = TypeKind.Delegate Then
-                        Return (typeSymbol, 0)
+                        Return (typeSymbol, 0, insertPosition)
                     End If
                 End If
             End If
 
             Dim symbolInfo = semanticModel.GetSymbolInfo(matchingNode, cancellationToken)
-            Return (If(symbolInfo.Symbol, symbolInfo.CandidateSymbols.FirstOrDefault()), 0)
+            Return (If(symbolInfo.Symbol, symbolInfo.CandidateSymbols.FirstOrDefault()), 0, insertPosition)
+        End Function
+
+        Private Shared Function TryGetInsertPositionFromDeclaration(matchingNode As SyntaxNode) As Integer
+            Dim parameters = matchingNode.ChildNodes().OfType(Of ParameterListSyntax)().SingleOrDefault()
```

Comment:
SingleOrDefault will throw wih 3 parametrs.  that seems bad here.

Reply:
We are looking not for `ParameterSyntax `but for `ParameterListSyntax`. This is copied from existing code just some lines below: https://github.com/dotnet/roslyn/pull/38748/files#diff-eba1bdf6a70b1c9058a37a7b879a66dcR144

Do you see a problem here?